README.md

```diff
@@ -12,73 +12,63 @@ Configuration parsing should be easy as it sounds - Hence;
  * No implicit requirements at user site.
  * Can write the config back to key value pairs, given the same config description. 
    Allows user to generate and populate configurations in the system-env/property-files in a typesafe way from outside, ensuring successful parsing in the app.
- * Automatic description and report generation on the config variables.
+ * Automatic report generation on the config variables.
+ * Automatic man pages based on the config description,  Higher level documentation for sections of config - all with a single composable syntax.
  * Can accumulate maximum errors.
  * Insanely simple to use
 
 
 ## Usage
 
 ```scala
+package zio.config.examples
 
-import zio.{App, ZIO}
-import zio.config._
+import zio.config._, Config._
 import zio.console.Console
+import zio.{ App, ZIO }
+
+case class Prod(ldap: String, port: Int, dburl: Option[String])
+
+object Prod {
+  val prodConfig: ConfigDescriptor[Prod] =
+    (string("LDAP") <*> int("PORT") ~ "Example: 8888"<*>
+      string("DB_URL").optional ~ "Db related config")(Prod.apply, Prod.unapply) ~ "Prod Config"
+
+  val myAppLogic: ZIO[Config[Prod], Throwable, (String, Option[String])] =
+    for {
+      prod    <- config[Prod]
+      written <- write(Prod.prodConfig).run.provide(prod)
+      _       <- ZIO.effect(println(written))
+      _       <- ZIO.effect(println(userManual(Prod.prodConfig)))
+    } yield (prod.ldap, prod.dburl)
+}
 
 object ReadConfig extends App {
-  case class Prod(ldap: String, dburl: Option[String])
-
-  private val config =
-    (string("LDAP") <*>
-      string("DB_URL").optional)(Prod.apply, Prod.unapply)
-
-  // In real, this comes from environment
-  private val validConfig =
-    Map(
-      "LDAP"    -> "v1",
-      "DB_URL"  -> "v2"
-    )
-    
-  val myAppLogic: ZIO[Console with ConfigSource, ReadErrors, Unit] =
-    ZIO.accessM(env =>
-      for {
-        result <- read(config).run
-        (report, conf) = result
-        _ <- env.console.putStrLn(report.toString)
-        _ <- env.console.putStrLn(conf.toString)
-        map <- write(config).run.provide(conf).either
-        _ <- env.console.putStrLn(map.toString)
-      } yield ()
-    )
-
-  override def run(args: List[String]): ZIO[ReadConfig.Environment, Nothing, Int] = {
-    myAppLogic
-    .provide(ProgramEnv(mapSource(validConfig).configService))
-    .fold(_ => 1, _ => 0)
-  }
-}
 
-case class ProgramEnv(configService: ConfigSource.Service) extends ConfigSource with Console.Live
+  override def run(args: List[String]): ZIO[ReadConfig.Environment, Nothing, Int] =
+    Config
+      .fromEnv(Prod.prodConfig)
+      .flatMap(config => Prod.myAppLogic.provide(config))
+      .foldM(failure => ZIO.effectTotal(println(failure)) *> ZIO.succeed(1), _ => ZIO.succeed(0))
```

Comment:
`println(failure)`is it using the zio console?
if not needed I would suggest to use `fold(_ => 1, _ => 0)`

Reply:
Yes. Good catch. I was running this example and wanted to see if it accumulates error. Not sure if we need to keep this,
Incase those who would like to see the reason for the error?

Reply:
Oops. Yes I should use zio console. Thanks @wi101

Reply:
Oops. Yes I should use zio console. Thanks @wi101

Reply:
@wi101  Done!